src/main/kotlin/com/mparticle/kits/RoktKit.kt

```diff
@@ -318,6 +320,45 @@ class RoktKit : KitIntegration(), CommerceListener, IdentityListener, RoktListen
         }
     }
 
+    private fun verifyHashedEmail(attributes: MutableMap<String, String>?) {
+        if (attributes == null) return
+
+        val emailKey = MParticle.IdentityType.Email.name.lowercase()
+        val otherKey = MParticle.IdentityType.Other.name.lowercase()
+        val emailShaKey = "emailsha256"
+
+        val emailShaValue = attributes.entries.find { it.key.equals(emailShaKey, ignoreCase = true) }?.value
+        val otherValue = attributes.entries.find { it.key.equals(otherKey, ignoreCase = true) }?.value
+
+        when {
+            !emailShaValue.isNullOrEmpty() -> {
+                // If emailsha256 is already present, remove entries with email and other keys
+                val iterator = attributes.entries.iterator()
+                while (iterator.hasNext()) {
+                    val entry = iterator.next()
+                    if (entry.key.equals(emailKey, ignoreCase = true) ||
+                        entry.key.equals(otherKey, ignoreCase = true)) {
+                        iterator.remove()
+                    }
+                }
+            }
+
+            !otherValue.isNullOrEmpty() -> {
```

Comment:
So this isn't necessarily wrong but I think the we each implemented it will have a slightly different result.

With the way I did it, if both 'other' and 'emailsha256' are set the 'emailsha256' value is the final value for 'emailsha256'  https://github.com/mparticle-integrations/mparticle-apple-integration-rokt/pull/22/files 

Meanwhile if both are set in yours the final value for 'emailsha256' will be the 'other' value. Which one of these is coreect @rmi22186 ?

Reply:
If both `other` and `emailsha256` are passed to us via MP.rok.selectPlacements, then `emailsha256` should take priority because they are passing that directly to us.